### PR TITLE
Leverage the `signature` package for signing

### PR DESCRIPTION
**Pull request by dekkagaijin**
Signed-off-by: Jake Sanders <jsand@google.com>

### CODE REVIEW

**Comment by lukehinds** [comment]
I will try and test this and review later today, just need to resolve some CA connection issues I have.

File: cmd/sign.go [hunk 1]
```diff
@@ -78,20 +76,21 @@ var signCmd = &cobra.Command{
...
-		pubKey := pub.(*ecdsa.PublicKey)
-		pubBytes, err := x509.MarshalPKIXPublicKey(pubKey)
+		pubBytes, err := x509.MarshalPKIXPublicKey(pub)
 		if err != nil {
 			return err
 		}
 
-		emailHash := sha256.Sum256([]byte(email))
-		proof, err := ecdsa.SignASN1(rand.Reader, privKey, emailHash[:])
+		proof, _, err := signer.Sign(ctx, []byte(email))
```
Comment (lukehinds) [inline cmd/sign.go:93]: Does signer.Sign also generate a hash as well as sign?

Comment (dlorenc) [reply cmd/sign.go:93]: Yeah, my understanding is that it's "signature, signed, error", some algorithms (ed25519) do hashing on their own, others require you to prehash, some like fido2 actually mutate the input payload in other ways before signing.

Comment (lukehinds) [reply cmd/sign.go:93]: ok, got it now.

**Review by lukehinds — APPROVED** [review]

File: cmd/sign.go (hunk 1 above)
Comment (dekkagaijin) [reply cmd/sign.go:93]: Yup, what Dan said.

**Pull request merged**